pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -238,11 +235,28 @@ func (s *ServiceEntryStore) serviceEntryHandler(old, curr model.Config, event mo
 	}
 
 	if fullPush {
+		// When doing a full push, for added and updated services trigger an eds update
+		// so that endpoint shards are updated.
+		var instances []*model.ServiceInstance
+		if len(addedSvcs) > 0 {
+			instances = append(instances, convertInstances(curr, addedSvcs)...)
+		}
+		if len(updatedSvcs) > 0 {
+			instances = append(instances, convertInstances(curr, updatedSvcs)...)
+		}
+		s.edsUpdate(instances)
+
+		// If service entry is deleted, cleanup endpoint shards for services.
+		for _, svc := range deletedSvcs {
+			s.XdsUpdater.SvcUpdate(s.Cluster(), string(svc.Hostname), svc.Attributes.Namespace, event)
```

Comment:
There was a bug: missing SvcUpdate on added/updated svc as kube controller does

Reply:
did not understand - do you mean it got fixed now?

Reply:
I mean we should also call this for  added/updated svc

Reply:
https://github.com/istio/istio/blob/master/pilot/pkg/xds/eds.go#L142 - It does nothing for added/updated. But no harm in calling. I will add

Reply:
There is just a counter